tool/tctl/common/plugin/entraid.go

```diff
@@ -0,0 +1,401 @@
+// in Teleport and a Teleport plugin to synchronize access lists from EntraID to Teleport.
+func (p *PluginsCommand) InstallEntra(ctx context.Context, args installPluginArgs) error {
+	inputs := p.install
+
+	proxyPublicAddr, err := getProxyPublicAddr(ctx, args.authClient)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	settings, err := p.entraSetupGuide(proxyPublicAddr)
+	if err != nil {
+		if errors.Is(err, errCancel) {
+			return nil
+		}
+		return trace.Wrap(err)
+	}
+
+	var tagSyncSettings *types.PluginEntraIDAccessGraphSettings
+	if settings.accessGraphCache != nil {
+		tagSyncSettings = &types.PluginEntraIDAccessGraphSettings{
+			AppSsoSettingsCache: settings.accessGraphCache.AppSsoSettingsCache,
+		}
+	}
+
+	saml, err := types.NewSAMLConnector(inputs.entraID.authConnectorName, types.SAMLConnectorSpecV2{
+		AssertionConsumerService: strings.TrimRight(proxyPublicAddr, "/") + "/v1/webapi/saml/acs/" + inputs.entraID.authConnectorName,
+		AllowIDPInitiated:        true,
+		// AttributesToRoles is required, but Entra ID does not have a default group (like Okta's "Everyone"),
+		// so we add a dummy claim that will never be fulfilled with the default configuration instead,
+		// and expect the user to modify it per their requirements.
+		AttributesToRoles: []types.AttributeMapping{
+			{
+				Name:  "https://example.com/my_attribute",
+				Value: "my_value",
+				Roles: []string{"requester"},
+			},
+		},
+		Display:             "Entra ID",
+		EntityDescriptorURL: entraapiutils.FederationMetadataURL(settings.tenantID, settings.clientID),
+	})
+	if err != nil {
+		return trace.Wrap(err, "failed to create SAML connector")
+	}
+
+	if _, err = args.authClient.CreateSAMLConnector(ctx, saml); err != nil {
+		if !trace.IsAlreadyExists(err) || !inputs.entraID.force {
+			return trace.Wrap(err, "failed to create SAML connector")
+		}
+		if _, err = args.authClient.UpsertSAMLConnector(ctx, saml); err != nil {
+			return trace.Wrap(err, "failed to upsert SAML connector")
+		}
+	}
+
+	if !inputs.entraID.useSystemCredentials {
+		integrationSpec, err := types.NewIntegrationAzureOIDC(
+			types.Metadata{Name: inputs.name},
+			&types.AzureOIDCIntegrationSpecV1{
+				TenantID: settings.tenantID,
+				ClientID: settings.clientID,
+			},
+		)
+		if err != nil {
+			return trace.Wrap(err, "failed to create Azure OIDC integration")
+		}
+
+		if _, err = args.authClient.CreateIntegration(ctx, integrationSpec); err != nil {
+			if !trace.IsAlreadyExists(err) || !inputs.entraID.force {
+				return trace.Wrap(err, "failed to create Azure OIDC integration")
+			}
+			if err = args.authClient.DeleteIntegration(ctx, integrationSpec.GetName()); err != nil {
+				return trace.Wrap(err, "failed to delete Azure OIDC integration")
+			}
+			if _, err = args.authClient.CreateIntegration(ctx, integrationSpec); err != nil {
+				return trace.Wrap(err, "failed to create Azure OIDC integration")
+			}
```

Comment:
This is kinda awkward, we need proper UpdateIntegration and UpdatePlugin APIs if we don't have one already. Outside of scope of this one though.